base/Enums.jl

```diff
@@ -5,12 +5,14 @@ module Enums
 import Core.Intrinsics.box
 export Enum, @enum
 
-abstract Enum
+function basetype end
 
-Base.convert{T<:Integer}(::Type{T}, x::Enum) = convert(T, box(Int32, x))
+abstract Enum{T<:Integer}
```

Comment:
I guess we need to think about whether or not it makes sense to have `Enum` be a parametric type.